### DISCUSSION TITLE
A question on Projecting domain wall state to zero momentum sector with + parity

### DISCUSSION THREAD

**Discussion by Sourav-physics** [QuSpin/QuSpin#457] [Q&A] [answered]
Hi, I have a sort of very naive query as follows.

I have an Ising system (with both transverse and longitudinal) field which is subjected to periodic drive. Now I am studying dynamics in the zero momentum sector with positive parity. My  initial state is domain wall state which I need to project in the zero momentum subspace with + parity. I am following the the code snippet as:

#------basis construction --------------#

basis = spin_basis_1d ( L = L, kblock = 0, pblock = 1 )
basis_full = spin_basis_1d ( L = args.L)

#----projection of initial state-------------#

CDW = "".join("1" for i in range(args.L//2)) + "".join("0" for i in range(args.L-args.L//2))
psi_full = np.zeros(basis_full.Ns) 
psi_full[ basis_full.index(CDW) ] = 1.0
P = basis.get_proj(dtype = np.float64) #-------Projector-------------------#
psi_0 = P.T.dot (psi_full)

However, I am not sure if the above approach is at all correct. Particularly when I attempt to calculate the half-chain entanglement entropy using the basis.ent_entanglement of the time evolved wavefunction (subjected to periodic drive) from the above obtained initial state, I get wrong results. I assume my projection is not correct. If so, could you please let me know how to project the initial domain wall state to the zero momentum sector with + parity correctly so that I can perform the dynamics starting from that state?

**Comment by mgbukov** [comment, marked as answer]
I dont immediately see anything obviously wrong. 

If I were you, I'd check if the dynamics is relevant at all here, i.e. if you get the wrong results already at time=0. 

Here're a few things to consider:
i) is `psi_0` normalized?
ii) momentum and parity do not commute; what one does to circumvent this is to use semimomentum states -- you have to see if they mess up the projection. One way to see if the problem might be coming from this, is to use only one of the two symmetries and check to see if you get the correct results with a single symmetry. See here for more [details on semimomentum states](https://github.com/weinbe58/QuSpin/issues/311).

**Reply by Sourav-physics** [reply]
Okay. I understand. I used only one symmetry and put kblock = 0. Indeed, after the projection operation is done, psi_0 does not remain normalized. So I normalized it (though I am not sure if it is sensible enough to do so). I can now see that at t=0, there is an entanglement entropy. So it means that after projecting the domain wall state (which is a product state with 0 entanglement entropy), I end up with a state which is mixed. But I am not sure if it is evident or expected.

**Comment by mgbukov** [comment]
> Okay. I understand. I used only one symmetry and put kblock = 0. Indeed, after the projection operation is done, psi_0 does not remain normalized. So I normalized it (though I am not sure if it is sensible enough to do so). I can now see that at t=0, there is an entanglement entropy. So it means that after projecting the domain wall state (which is a product state with 0 entanglement entropy), I end up with a state which is mixed. But I am not sure if it is evident or expected.

It is OK to end up with an entangled pure state after the projection. To see this, take just two spins and translation symmetry. The k=0 manifold is the triplet manifold (3 states), while the k=1 manifold is the singlet manifold (remaining one state). In the singlet manifold, the state is the singlet ~|01> - |10>. This is a maximally-entangled Bell state. So using symmetries makes some of your basis  states entangled in general. 

In order to do simulation properly, what you have to do is project down to the symmetry sector (keeping the new normalization constant), do the evolution, and then go back to the full basis (weighing the different contributions by these normalization constants). Only then can u compute Sent and you have to do it in the full basis. 

Going back and forth between bases can be done using `basis.project_from()` and `basis.project_to()` -- which do not compute the entire projector.

**Reply by Sourav-physics** [reply]
Ok, I understand. Thanks a lot :-)
